### ISSUE TITLE
Trait auto import quick fix not triggered by impl traits

### ISSUE THREAD

**Issue by agluszak** [intellij-rust/intellij-rust#9111]
## Environment

* **IntelliJ Rust plugin version:** 0.4.175.4772-221
* **Rust toolchain version:** 1.62.0 (a8314ef7d 2022-06-27) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2022.1.3 (CL-221.5921.27)
* **Operating system:** Linux 5.17.15-76051715-generic
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
Trait auto import quick fix is not triggered for `impl Trait` values. 

Related to  (I guess?) #2774 and #2740

I'll try to fix it myself :)

## Steps to reproduce
`cargo install warp@0.3`
```rust
use std::fs::File;
use std::io::Read;
use warp::hyper::Body;


async fn repro() -> Result<(), warp::hyper::Error> {
    let body = Body::empty();
    let body_reader = warp::hyper::body::aggregate(body).await?.reader(); // no suggestion
    Ok(())
}

fn main() {
    let mut s = String::new();
    let mut f = File::open("src/main.rs").unwrap();
    f.read_to_string(&mut s).unwrap(); // that works
    println!("Hello, world!");
}

```
```
cargo build                                                                                                                                                                                                                                                                           18:11:34
   Compiling asdasd v0.1.0 (/home/andrzej.gluszak/code/asdasd)
error[E0599]: no method named `reader` found for opaque type `impl Buf` in the current scope
   --> src/main.rs:8:65
    |
8   |     let body_reader = warp::hyper::body::aggregate(body).await?.reader(); // no suggestion
    |                                                                 ^^^^^^ method not found in `impl Buf`
    |
   ::: /home/andrzej.gluszak/.cargo/registry/src/github.com-1ecc6299db9ec823/bytes-1.2.0/src/buf/buf_impl.rs:900:8
    |
900 |     fn reader(self) -> Reader<Self>
    |        ------ the method is available for `impl Buf` here
    |
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
1   | use warp::Buf;
    |

For more information about this error, try `rustc --explain E0599`.

```

or even simpler (that confirms my theory that it's about `impl Traits`:

```rust
mod inner {
    pub trait Foo {
        fn foo(&self);
    }

    struct Bar;

    impl Foo for Bar {
        fn foo(&self) {}
    }

    pub fn foo() -> impl Foo {
        Bar
    }
}

fn main() {
    let test = inner::foo();
    test.foo(); // no suggestion
}
```

**Comment by laralove143** [comment]
This is really tedious. It doesn't even work for `std`'s traits like `std::str::FromStr`
